include/openthread/history_tracker.h

```diff
@@ -0,0 +1,211 @@
+ * @param[inout] aIterator   A pointer to an iterator. MUST be initialized or the behavior is undefined.
+ * @param[out]   aEntryAge   A pointer to a variable to output the entry's age. MUST NOT be NULL.
+ *                           Age is provided as the duration (in milliseconds) from when entry was recorded to
+ *                           @p aIterator initialization time. It is set to `OT_HISTORY_TRACKER_MAX_AGE` for entries
+ *                           older than max age.
+ *
+ * @returns The `otHistoryTrackerMessageInfo` entry or `NULL` if no more entries in the list.
+ *
+ */
+const otHistoryTrackerMessageInfo *otHistoryTrackerIterateRxHistory(otInstance *              aInstance,
+                                                                    otHistoryTrackerIterator *aIterator,
+                                                                    uint32_t *                aEntryAge);
+
+/**
+ * This function iterates over the entries in the TX message history list.
+ *
+ * @param[in]    aInstance   A pointer to the OpenThread instance.
+ * @param[inout] aIterator   A pointer to an iterator. MUST be initialized or the behavior is undefined.
+ * @param[out]   aEntryAge   A pointer to a variable to output the entry's age. MUST NOT be NULL.
+ *                           Age is provided as the duration (in milliseconds) from when entry was recorded to
+ *                           @p aIterator initialization time. It is set to `OT_HISTORY_TRACKER_MAX_AGE` for entries
+ *                           older than max age.
+ *
+ * @returns The `otHistoryTrackerMessageInfo` entry or `NULL` if no more entries in the list.
+ *
+ */
+const otHistoryTrackerMessageInfo *otHistoryTrackerIterateTxHistory(otInstance *              aInstance,
+                                                                    otHistoryTrackerIterator *aIterator,
+                                                                    uint32_t *                aEntryAge);
+
+#define OT_HISTORY_TRACKER_ENTRY_AGE_STRING_SIZE 25 ///< Recommended size for string representation of an entry age.
+
+/**
+ * This function converts a given entry age to a human-readable string.
+ *
+ * The entry age string follows the format "<hh>:<mm>:<ss>.<mmmm> ago" for hours, minutes, seconds and millisecond (if
+ * shorter than one day) or "<dd> days <hh>:<mm>:<ss>.<mmmm> ago" (if longer than one day).
+ *
+ * If the resulting string does not fit in @p aBuffer (within its @p aSize characters), the string will be truncated
+ * but the outputted string is always null-terminated.
+ *
+ * @param[in]  aEntryAge The entry age (duration in msec).
+ * @param[out] aBuffer   A pointer to a char array to output the string (MUST NOT be NULL).
+ * @param[in]  aSize     The size of @p aBuffer (in bytes). Recommended to use `OT_IP6_ADDRESS_STRING_SIZE`.
+ *
+ */
+void otHistoryTrackerEntryAgeToString(uint32_t aEntryAge, char *aBuffer, uint16_t aSize);
```

Comment:
I am not sure if this will be useful. This will restrict the API use (caller need to give an exact array size or do pointer array casting which is tricky and can be error-prone).
- We want to allow user to give us any buffer with whatever size they want (e.g. user may have a large static buffer which they use for all string conversions).
- Other public APIs that convert to string accept any buffer.

Reply:
> We want to allow user to give us any buffer with whatever size they want (e.g. user may have a large static buffer which they use for all string conversions).

A larger static buffer can be converted to a `char (*aBuffer)[OT_HISTORY_TRACKER_ENTRY_AGE_STRING_SIZE])` safely, even though it requires some skill of pointer manipulation.

The suggested code can help reduce code size. But fine to me to keep it as is.